barretenberg/cpp/src/barretenberg/vm2/common/protocol_contracts.hpp

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "barretenberg/vm2/common/aztec_types.hpp"
+#include "barretenberg/vm2/common/protocol_contract_data.hpp"
+
+#include <cassert>
+
+namespace bb::avm2 {
+
+inline bool is_protocol_contract(AztecAddress canonical_address)
+{
+    return derived_addresses.find(canonical_address) != derived_addresses.end();
+}
+
+inline AztecAddress get_derived_address(AztecAddress canonical_address)
```

Comment:
- take as const&
- you might be able to return const& as well given you take it from the dict